src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -131,17 +134,23 @@ class ClusterMonitorActor(val cluster: Cluster,
 
     // Then remove the Dataproc Worker IAM role for the pet service account
     // Only do this if the cluster was created with the pet service account.
-    val iamFuture = deleteBucketFuture flatMap { _ => removeIamRolesForUser() }
+    val iamFuture = deleteInitBucketFuture flatMap { _ => removeIamRolesForUser() }
```

Comment:
FYI these three future-vals (`iamFuture`, `bucketACLFuture` and `dbFuture`) are going to start executing as soon as they're created (i.e. in parallel). if you want them to execute sequentially you can either 

1. chain the flatMaps (like before)
2. move their bodies into the for-comp
3. define them as functions and call the functions in the for-comp

Reply:
Good point.  I think it's fine for the first 2 to run in parallel.  I'll move the DB update so it waits for them.